nemo/collections/nlp/modules/common/megatron/megatron_encoder_decoder.py

```diff
@@ -171,7 +171,7 @@ def forward(
         # decoder
         # Adjust encoder attention mask if encoder is a perceiver.
         if self.encoder is not None and isinstance(self.encoder, MegatronPerceiverEncoderModule):
-            enc_attn_mask = torch.ones(enc_output.size(0), self.hidden_steps).to(enc_output.device)
+            enc_attn_mask = torch.ones(enc_output.size(1), self.hidden_steps).to(enc_output.device)
```

Comment:
Here we expect [S x B x H]  right? Can we add it in comment?

Reply:
The expected shape of the mask is [B x S], but `enc_output` is now [S x B x H], so I need size(1) to get the batch size.